src/com/owncloud/android/files/InstantUploadBroadcastReceiver.java

```diff
@@ -184,6 +197,11 @@ private void handleConnectivityAction(Context context, Intent intent) {
             Cursor c = db.getAwaitingFiles();
             if (c.moveToFirst()) {
                 do {
+                    if (instantPictureUploadViaWiFiOnly(context) &&
+                            !isConnectedViaWiFi(context)){
+                        break;
```

Comment:
At this point, can you be sure that the current upload is an instant picture upload?
What would happen if `instantPictureUploadViaWiFiOnly()==true` and `isConnectedViaWiFi()==false` but the users just decided to upload, e.g., their shopping list?